src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseStudentParticipationRepository.java

```diff
@@ -48,6 +48,17 @@ SELECT max(prr.id)
     Optional<ProgrammingExerciseStudentParticipation> findByIdWithLatestResultAndFeedbacksAndRelatedSubmissions(@Param("participationId") Long participationId,
             @Param("dateTime") ZonedDateTime dateTime);
 
+    @Query("""
+            SELECT DISTINCT p
+            FROM ProgrammingExerciseStudentParticipation p
+                LEFT JOIN FETCH p.results pr
+                LEFT JOIN FETCH pr.feedbacks f
+                LEFT JOIN FETCH f.testCase
+                LEFT JOIN FETCH p.submissions
+            WHERE p.id = :participationId
+             """)
+    Optional<ProgrammingExerciseStudentParticipation> findByIdWithAllResultsAndFeedbacksAndRelatedSubmissions(@Param("participationId") Long participationId);
```

Comment:
Do we really need all feedback elements for ALL results at one time? Keep in mind that some students use Artemis-driven-development, leading to 50 - 100 submissions and let's say up to 100 * 20 = 2000 feedback items. Loading them all at once is memory intensive and will take a while.

Hava a look at other places of Artemis where an overview for all results is displayed. Here we just load the result, without feedback or test cases. If the feedback needs to be shown, an API-call to the `/details` endpoint is made.

Reply:
Thank you! I just checked and they weren't necessary as you said